modules/caddyhttp/reverseproxy/forwardauth/caddyfile.go

```diff
@@ -196,9 +195,7 @@ func parseCaddyfile(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error)
 	// need at least one handler in the routes for the response handling
 	// logic in reverse_proxy to not skip this entry as empty.
 	for from, to := range headersToCopy {
-		handler.Request.Set[to] = []string{
-			"{http.reverse_proxy.header." + from + "}",
-		}
+		handler.Request.Set.Set(to, "{http.reverse_proxy.header."+http.CanonicalHeaderKey(from)+"}")
```

Comment:
I guess we can canonicalize both sides? Maybe `.Set()` already does that for us though?

```suggestion
		handler.Request.Set.Set(http.CanonicalHeaderKey(to), "{http.reverse_proxy.header."+http.CanonicalHeaderKey(from)+"}")
```

Reply:
Yep, the header methods do the canonicalization for us: https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/net/http/header.go;l=24-60